crates/turborepo-lib/src/run/scope/filter.rs

```diff
@@ -38,7 +38,7 @@ impl PackageInference {
         );
         let full_inference_path = turbo_root.resolve(pkg_inference_path);
         for (workspace_name, workspace_entry) in pkg_graph.workspaces() {
-            let pkg_path = turbo_root.resolve(workspace_entry.package_json_path());
+            let pkg_path = turbo_root.resolve(workspace_entry.package_path());
```

Comment:
Previously we would be checking if `/path/to/repo/packages/foo/package.json` contained a directory which would always be false.